### PR TITLE
gci: fix issues and re-enable autofix

### LINKED ISSUE #2604: Add autofix for gci
### Your feature request related to a problem? Please describe.

We were using the GCI linter to enforce import order. However, with the 1.44.2 update, it seems like auto-fix support was dropped.
This change makes it harder to enforce import order as the developer has to manually fix the code.

### Describe the solution you'd like.

It seems like GCI supports write mode. Using it for auto fix could be possible.

### Describe alternatives you've considered.

Removing the GCI linter as it is currently unusable, or downgrading golangci-lint.

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by daixiang0**
https://github.com/daixiang0/gci/compare/v0.3.3...v0.3.4

Fixes #2604
Close #2653

### CODE REVIEW

**Comment by daixiang0** [comment]
Sorry about breaking for some time, now local tests all pass, and auto fix works.

File: go.mod [hunk 1]
```diff
@@ -20,7 +20,7 @@ require (
 	github.com/breml/errchkjson v0.3.0
 	github.com/butuzov/ireturn v0.1.1
 	github.com/charithe/durationcheck v0.0.9
-	github.com/daixiang0/gci v0.3.3
+	github.com/daixiang0/gci v0.3.4-0.20220530072138-bf921fe4731b
```
Comment (ldez) [inline go.mod:23]: a tag is required.

**Review by ldez — CHANGES_REQUESTED** [review]

File: pkg/golinters/gci.go [hunk 2]
```diff
@@ -15,33 +18,121 @@ import (
...
+		rawCfg := gci.GciStringConfiguration{
+			Cfg: gcicfg.FormatterConfiguration{
+				NoInlineComments: settings.NoInlineComments,
+				NoPrefixComments: settings.NoPrefixComments,
+			},
+			SectionStrings:          settings.Sections,
+			SectionSeparatorStrings: settings.SectionSeparator,
 		}
 
 		if settings.LocalPrefixes != "" {
```
Comment (daixiang0) [inline pkg/golinters/gci.go:41]: I want to remove LocalPrefixes here since I do not support that anymore in golangci-lint. @ldez

Comment (ldez) [reply pkg/golinters/gci.go:41]: inside golancci-lint we don't want to break

**Review by ldez — APPROVED** [review]
LGTM

**Comment by schrej** [comment]
Since you've mentioned in Discussions that there isn't a release schedule and you base them on features: Would it be possible to get a release that includes this fix? It's soft-blocking the go1.18 upgrade in kubernetes-sigs/cluster-api.
There aren't too many changes from what I can [see](https://github.com/golangci/golangci-lint/compare/v1.46.2...master), so maybe a patch release would be suitable.

**Comment by ldez** [comment]
Hello, there are some minor features that have been merged, so we will not create a patch release.
I need to work on some fixes before creating a new minor release, I hope I will be able to do that soon.

**Pull request merged**